lib/runner/TestRunner.js

```diff
@@ -151,13 +159,26 @@ module.exports = class TestRunner {
             this.emit("onTestEnd");
 
         }
+        stopWatchTestSuite.stop();
 
-        await this.saveTestResults(testResults, testSuiteWithLocale, startTimeTestSuite, getSourceExistPromise);
-        await invoker.after(testSuiteWithLocale);
+        const customerId = await this.getCustomerIdFromSource(testSuiteWithLocale.virtualDeviceToken);
```

Comment:
why dont put this logic on the server?

Reply:
It is because the endpoint should be only responsible for saving data, not for business logic.

If your recommendation for moving this into the endpoint, just answer this comment and I will change it

Reply:
guess you can leave it like this, just add the validation when this method fails